src/components/animated/_ripple.less

```diff
@@ -0,0 +1,79 @@
+.ripple-btn {
+  .bc(transparent);
+  position: relative;
+
+  &::before {
+    content: " ";
+    z-index: -1;
+    position: absolute;
+    height: 0;
+    width: 0;
+    top: 50%;
+    left: 50%;
+    border-radius: 50%;
+    transform: translate(-50%, -50%);
+  }
+
+  &:hover {
+    color: white !important;
+    transition: 0.5s;
+  }
+
+  &:hover::before {
+    animation: a-ripple 1s cubic-bezier(0.39, 0.58, 0.57, 1) forwards;
+  }
+
+  .rippleBtn(@color) {
+    border-color: @color;
+    color: @color;
+
+    &::before {
+      .bc(@color);
+    }
+  }
+
+  &.blue-btn {
+    .rippleBtn(@blue);
+  }
+  &.pink-btn {
+    .rippleBtn(@pink);
+  }
+
+  &.green-btn {
+    .rippleBtn(@green);
+  }
+
+  &.yellow-btn {
+    .rippleBtn(@yellow);
```

Comment:
Can you add for yellow something similar to in "fill-btn" where before hover the color of text is yellow, but then on hover it becomes black? You can use the example from there.